### PR TITLE
fix(ecs): iam policies are missing for firelens cloudwatch_logs log driver

### PR DESCRIPTION
**Pull request by rhermes62**
We recently migrated our FireLens log driver from using `cloudwatch` to using the recommended `cloudwatch_logs`.

`cloudwatch_logs` is the newer and more high performance version as detailed in this guidance from AWS https://github.com/aws/amazon-cloudwatch-logs-for-fluent-bit#new-higher-performance-core-fluent-bit-plugin

However, when doing that, we lost the automatic configuration of the IAM policies to write to CloudWatch which broke our logging.

Looks like the CDK only auto adds CloudWatch IAM policies only for the `cloudwatch` plugin opposed to both the older `cloudwatch` plugin and newer and recommended `cloudwatch_logs` plugin.

Policies should be added to both so that customer's logging doesn't break when migrating to the newer plugin.

----

### All Submissions:

* [ ✅] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ecs/test/firelens-log-driver.test.ts [hunk 1]
```diff
@@ -189,7 +189,8 @@ describe('firelens log driver', () => {
     });
 
     // THEN
-    Template.fromStack(stack).hasResourceProperties('AWS::ECS::TaskDefinition', {
+    const template = Template.fromStack(stack)
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-ecs/test/firelens-log-driver.test.ts:192]:
```suggestion
    const template = Template.fromStack(stack);
```

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
First let's get this build succeeding. I put a correction below that is causing the failure. 

Additionally, please make sure that your PR title confirms to the conventional commit standard (fix, feat, chore) and that it is written in a style that will reflect correctly in the change log (See [Contributing Guide, Pull Requests](https://github.com/aws/aws-cdk/blob/v1-main/CONTRIBUTING.md#step-4-pull-request)). A fix should state the problem that is being addressed, not the solution being implemented.

We'll provide further feedback once we get a working build.

**Review by TheRealAmazonKendra — DISMISSED** [review]
This looks good but I'm concerned that this issue wasn't caught by our current integ test coverage. I think we need an integration test for this use case. Can you please add one?

**Comment by rhermes62** [comment]
Works with me! None of these plugins have integ coverage AFAIK.

I will add for the `cloudwatch` and `cloudwatch_logs` plugins since those are the primary focus of the PR.

Should I also add them for the `firehose` and `kinesis` options which are also missing integ coverage? Or is that scope creep/more suitable for a separate chore PR?

**Comment by TheRealAmazonKendra** [comment]
> Works with me! None of these plugins have integ coverage AFAIK.
> 
> I will add for the `cloudwatch` and `cloudwatch_logs` plugins since those are the primary focus of the PR.
> 
> Should I also add them for the `firehose` and `kinesis` options which are also missing integ coverage? Or is that scope creep/more suitable for a separate chore PR?

I will never turn down extra test coverage! However, let's do the extra ones in a separate PR so we can get this one merged faster.

**Comment by TheRealAmazonKendra** [comment]
@rhermes62 Just to clarify, I'd like the tests for `cloudwatch` and `cloudwatch_logs` added to this PR and the others in a separate PR. After reading my own response, I wasn't sure if it sounded like I wanted them all in a separate PR or not.

**Comment by rhermes62** [comment]
Makes sense and will do! Will try and squeeze in those extra tests tonight or tomorrow

**Comment by rhermes62** [comment]
Still working on getting those cleaned up. Builds in flight likely to fail.

These first integ tests for aws-cdk written so learning how this repo does them. I had historically always just used jest's snapshotting feature + the aws-cdk-lib/assertions's Template feature to write snapshot tests like `expect(Template.fromStack(stack).toJSON()).toMatchSnapshot();`.

**Comment by TheRealAmazonKendra** [comment]
> Still working on getting those cleaned up. Builds in flight likely to fail.
> 
> These first integ tests for aws-cdk written so learning how this repo does them. I had historically always just used jest's snapshotting feature + the aws-cdk-lib/assertions's Template feature to write snapshot tests like `expect(Template.fromStack(stack).toJSON()).toMatchSnapshot();`.

No worries! If you need any help troubleshooting, please let me know. I'm happy to assist.

**Comment by TheRealAmazonKendra** [comment]
@rhermes62 Actually, I just took a look and I can make myself useful here immediately. Just run `yarn integ --update-on-failed integ.firelens-log-driver.js` and it will update your test folder to the correct name. I ran it locally and the new test succeeds so I don't think you should have any issue on your end.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Just putting this back in changes requested so I see when that update has been made.

**Comment by rhermes62** [comment]
Sounds good! Thank you! We normally don't have direct access/credentials to do things like `cdk deploy` to AWS accounts since we have strict release processes/security so I was trying to find one to deploy to😂😅

I was about to create a temp account just to do test deploy to in order to generate the artifact files to get the integ to pass.

**Comment by TheRealAmazonKendra** [comment]
> Sounds good! Thank you! We normally don't have direct access/credentials to do things like `cdk deploy` to AWS accounts since we have strict release processes/security so I was trying to find one to deploy to😂😅
> 
> I was about to create a temp account just to do test deploy to in order to generate the artifact files to get the integ to pass.

The test will need to deploy when you do it that way. It actually runs the test with deployment and deletion of resources. If you have issues doing that. I can upload the output for you. Just let me know if that's needed.

**Pull request closed without merging**